examples/cockroachdb/cockroachdb-petset.yaml

```diff
@@ -0,0 +1,88 @@
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+  name: cockroachdb
+  labels:
+    app: cockroachdb
+spec:
+  ports:
+  - port: 26257
+    targetPort: 26257
+    name: grpc
+  - port: 8080
+    targetPort: 8080
+    name: http
+  clusterIP: None
+  selector:
+    app: cockroachdb
+---
+apiVersion: apps/v1alpha1
+kind: PetSet
+metadata:
+  name: cockroachdb
+spec:
+  serviceName: "cockroachdb"
+  replicas: 5
+  template:
+    metadata:
+      labels:
+        app: cockroachdb
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      containers:
+      - name: cockroachdb
+        # Runs the master branch. Not recommended for production.
+        image: cockroachdb/cockroach:latest
+        ports:
+        - containerPort: 26257
+          name: grpc
+        - containerPort: 8080
+          name: http
+        volumeMounts:
+        - name: datadir
+          mountPath: /cockroach/cockroach-data
+        command:
+          - "/bin/bash"
+          - "-ecx"
+          - |
+            # The use of qualified `hostname -f` is crucial:
+            # Other nodes aren't able to look up the unqualified hostname.
+            CRARGS=("start" "--logtostderr" "--insecure" "--host" "$(hostname -f)")
+            # TODO(tschottdorf): really want to use an init container to do
+            # the bootstrapping. The idea is that the container would know
+            # whether it's on the first node and could check whether there's
+            # already a data directory. If not, it would bootstrap the cluster.
+            # We will need some version of `cockroach init` back for this to
+            # work. For now, just do the same in a shell snippet.
+            # Of course this isn't without danger - if node0 loses its data,
+            # it will upon restarting it will simply bootstrap a new cluster
+            # and smack it into our existing cluster.
+            # There are likely ways out. For example, the init container could
+            # query the kubernetes API and see whether any other nodes are
+            # around, etc. Or, of course, the admin can pre-seed the lost
+            # volume somehow (and in that case we should provide a better way,
+            # for example a marker file).
+            if [ ! "$(hostname)" == "cockroachdb-0" ] || \
+               [ -e "/cockroach/cockroach-data/COCKROACHDB_VERSION" ]
+            then
+              CRARGS+=("--join" "cockroachdb")
+            fi
+            /cockroach/cockroach ${CRARGS[*]}
+      volumes:
+      - name: datadir
```

Comment:
Yeah, but I'd rather do it with emptyDir.  I don't think tmpfs solves the
problem for medium sized directory sets.

Ultimately this is a storage class ("free") and a durability guarantee
("instance") and a quota (based on request capacity).  Since we already
have an alpha empty dir quota solution I want to sketch out the path that
gets us to where users *always* just create PVCs and they never have design
without them.

On Jul 4, 2016, at 9:06 PM, Prashanth B <notifications@github.com> wrote:

In examples/cockroachdb/cockroachdb-petset.yaml
<https://github.com/kubernetes/kubernetes/pull/28446#discussion_r69497879>:

> +            # Of course this isn't without danger - if node0 loses its data,
> +            # it will upon restarting it will simply bootstrap a new cluster
> +            # and smack it into our existing cluster.
> +            # There are likely ways out. For example, the init container could
> +            # query the kubernetes API and see whether any other nodes are
> +            # around, etc. Or, of course, the admin can pre-seed the lost
> +            # volume somehow (and in that case we should provide a better way,
> +            # for example a marker file).
> +            if [ ! "$(hostname)" == "cockroachdb-0" ] || \
> +               [ -e "/cockroach/cockroach-data/COCKROACHDB_VERSION" ]
> +            then
> +              CRARGS+=("--join" "cockroachdb")
> +            fi
> +            /cockroach/cockroach ${CRARGS[*]}
> +      volumes:
> +      - name: datadir

Or always deploy a default throwaway directory storage provisioner #27732
<https://github.com/kubernetes/kubernetes/issues/27732>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/28446/files/a402b08521bc41a113668d2f4f1fab5f10622302#r69497879>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p_fTiXcbkOHDG-WXI00TpfBCt2wnks5qSa4bgaJpZM4JEBSZ>
.